### ISSUE TITLE
BannedApiAnalyzers: Not able to ban string plus operator

### ISSUE THREAD

**Issue by inosik** [dotnet/roslyn-analyzers#7608]
### Analyzer

**Diagnostic ID**: [RS0030](https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.BannedApiAnalyzers/BannedApiAnalyzers.Help.md): `The symbol '...' is banned in this project.`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.BannedApiAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.BannedApiAnalyzers)

**Version**: 3.3.4

### Describe the bug

I'm not able to ban `String.op_Addition(string, object)`. Custom types work fine.

### Steps To Reproduce

1. Create a C# class library project

1. Add Microsoft.CodeAnalysis.BannedApiAnalyzers to the project

2. Add the following `BannedSymbols.txt` to the project:

    ``` text
    M:BannedApisDemo.T.op_Addition(BannedApisDemo.T,System.Object)
    M:System.String.op_Addition(System.String,System.Object)
    ```

3. Add this class to the project:

    ``` csharp
    namespace BannedApisDemo;

    public class Demo
    {
        public static string A => "foo" + new object();

        public static T B = new T() + new object();
    }

    public class T
    {
        public static T operator +(T left, object right) => throw null!;
    }
    ```

### Expected behavior

The value expression of `A` and `B` should produce a warning.

### Actual behavior

Only the value expression of `B` produces a warning:

![Image](https://github.com/user-attachments/assets/74f8255d-5853-4bee-8101-1c094a277fc5)

## Additional context

I'm trying to prevent such code from being written (or existing code to be refactored to something similar):

``` csharp
-string value = ...
+object value = ...
 // A bunch of code in between ...
 var stringValue = "prefix: " + someObject;
```

**Comment by inosik** [comment]
This analyzer was moved to dotnet/roslyn. Can anybody of the maintainers move the issue please?

**Comment by inosik** [comment]
Looks like `M:System.String.op_Addition(System.String,System.Object)` doesn't really exist. The `+` operator is compiled as a combination of `Object.ToString()` and `String.Concat(string, string)`.

**Comment by Frulfump** [comment]
Probably worth you manually moving i.e re-creating and linking to this issue from dotnet/roslyn

**Comment by inosik** [comment]
Yeah, or just closing. As per my comment above, I don't think this can be done using the BannedApiAnalyzer.
